cmake/bazel_to_cmake.py

```diff
@@ -0,0 +1,282 @@
+    global package_prefix
+    return trim_multiple_ruy_prefixes(f'ruy_{package_prefix}_{name}')
+
+
+def get_cmake_dep_target_name(name):
+    if name in external_targets:
+        return name
+    if name.startswith('$'):
+        # Happens for deps that are the result of expanding a select() that we
+        # have compiled to expanding a variable.
+        return name
+    if name.startswith('//'):
+        after_last_slash = name.split('/')[-1]
+        if not ':' in after_last_slash:
+            name = f'{name}:{after_last_slash}'
+        raw=name[2:].replace('/', '_').replace(':', '_')
+        return trim_multiple_ruy_prefixes(raw)
+    if name.startswith(':'):
+        name = name[1:]
+    return get_cmake_local_target_name(name)
+
+
+#
+# Functions implementing BUILD functions
+#
+
+
+def package(**kwargs):
+    pass
+
+
+def exports_files(*args):
+    pass
+
+
+def load(filename, *args):
+    if filename.startswith('@'):
+        return
+    elif filename.startswith(':'):
+        filename = os.path.join(bazel_package_dir, filename[1:])
+    elif filename.startswith('//'):
+        split = filename[2:].split(':')
+        filename = os.path.join(bazel_workspace_dir, split[0], split[1])
+
+    src_file_content = open(filename).read()
+    processed_file_content = preprocess_input_text(src_file_content)
+    exec(processed_file_content, globals(), globals())
+
+
+def config_setting(**kwargs):
+    # Nothing to do since our implementation of select() is based on parsing
+    # the names of config_settings, not looking deep into their actual
+    # implementation.
+    pass
+
+
+def filegroup(**kwargs):
+    pass
+
+
+def config_setting_group(**kwargs):
+    # See config_setting.
+    pass
+
+
+def bzl_library(**kwargs):
+    pass
+
+
+select_index = 0
+select_cache = {}
+
+
+def select(dict):
+    global select_index
+    global select_cache
+    global package_prefix
+    key = pickle.dumps(dict)
+    if key in select_cache:
+        select_name = select_cache[key]
+    else:
+        unique_values = list(
+            {*itertools.chain.from_iterable(dict.values())})
+        unique_values.sort()  # sorting ensures determinism, no spurious diffs
+        description = '_'.join(unique_values)
+        select_name = f'{package_prefix}_{select_index}_{description}'
+        select_name = select_name.replace('c++', 'cxx')
+        select_name = re.sub(r'[^a-zA-Z0-9]+', '_', select_name)
+        select_index = select_index + 1
+        select_cache[key] = select_name
+        generate_cmake_select(select_name, dict)
+
+    return [f'${{{select_name}}}']
+
+
+def generic_rule(rule_name, **kwargs):
+    print(f'{rule_name}(')
+    for key in kwargs.keys():
+        values = kwargs[key]
+        if type(values) is bool:
+            if values:
+                print(f'  {key.upper()}')
+                continue
+            else:
+                raise ValueError(
+                    'How do we specify FALSE boolean args in CMake?')
```

Comment:
```suggestion
            else:
                raise ValueError(
                    'Cannot specify FALSE boolean args in CMake')
```

Make the default false and leave them off is how we do it, but that requires that you control all the rules and guarantee that.

Reply:
ah ok!